sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -998,10 +998,27 @@ abstract class BinaryComparison extends BinaryOperator with Predicate {
     }
   }
 
-  protected lazy val ordering: Ordering[Any] = TypeUtils.getInterpretedOrdering(left.dataType)
+  protected lazy val ordering: Ordering[Any] = new Ordering[Any] {
+    override def compare(x: Any, y: Any): Int = {
+      (x, y) match {
+        // Handle comparison when the left side is a String and the right side is an Integer
+        case (xs: String, yi: Int) => xs.toIntOption match {
```

Comment:
by this logic "000001" will be converted to 1, so "000001" == 1 = true. I am not sure this is expected behavior. 
I guess always converting int to string is more predictable

Reply:
Thank you Masykus! 
I've updated the comparison logic to always convert the integer to a string before performing the comparison, you can check it in the latest commit!